herd_behaviour.nlogo

```diff
@@ -27,37 +29,87 @@ to setup
     set v-dir pseudo-random-vector
     set color blue
     set w 0
+    set g-dir [0 0]
   ]
   let number_informed_members (number_herd_members * ( 0.01 * (percent_informed)))
   ask n-of number_informed_members turtles [
     set w  weight
     set color red
+    set g-dir [125 125]
   ]
 end
 
 to go
   ask turtles[
-    ifelse count other turtles in-radius avoidance_range > 0 [
+    let base [0 0] ;; base of function calculated in every
+    let ci-dir array:from-list (list xcor ycor) ;; the position vector of actual turtle
+    ifelse any? other turtles in-radius avoidance_range [
       ;; avoid them
-      calculate-dir
+      ask turtles in-radius avoidance_range [
+        let cj-dir array:from-list (list xcor ycor)
+        set base plus-vectors base calculate-base ci-dir cj-dir;; add the dividing by absolute vector
+      ]
+      set temp minus-one base
+      set d-dir unit-vector temp
+      ;; for informed turtles, extra formula 3
+      if w != 0[
+          set d-dir unit-vector plus-vectors d-dir multiply-vector-number g-dir w
+      ]
     ][
-      if count other turtles in-radius following_range > 0[
-       ;;follow them
-        calculate-dir-with-w-g
+      if any? other turtles in-radius following_range [
+        ;;follow them
+        let v-sum [0 0]
+        ask turtles in-radius avoidance_range [
+          let cj-dir array:from-list (list xcor ycor)
+          set base plus-vectors base calculate-base ci-dir cj-dir;; add the dividing by absolute vector
+
+          set v-sum plus-vectors v-sum unit-vector v-dir
+        ]
+        set temp plus-vectors base v-sum
+        set d-dir unit-vector temp
+
+        ;; for informed turtles, extra formula 3
+        if w != 0[
+          set d-dir unit-vector plus-vectors d-dir multiply-vector-number g-dir w
+        ]
       ]
     ]
     show v-dir
     forward 1
+    ;; where set v-dir d-dir ?
   ]
   tick
 end
 
-to calculate-dir ;; formula 1 in paper
 
+to-report minus-one[arr] ;; calculates a vector * -1
+  array:set arr 0 ((array:item arr 0) * (-1))
+  array:set arr 1 ((array:item arr 1) * (-1))
+  report arr
+end
+
+to-report multiply-vector-number[arr num] ;; calculates a vector * -1
+  array:set arr 0 ((array:item arr 0) * num)
+  array:set arr 1 ((array:item arr 1) * num)
+  report arr
+end
+
+to-report minus-vectors[x y]
+  array:set x 0 ((array:item x 0) - (array:item y 0))
+  array:set x 1 ((array:item x 1) - (array:item y 1))
+  report x
 end
 
-to calculate-dir-with-v ;; formula 2 in paper
+to-report plus-vectors[x y]
+  array:set x 0 ((array:item x 0) + (array:item y 0))
+  array:set x 1 ((array:item x 1) + (array:item y 1))
+  report x
+end
 
+to-report calculate-base[ci-dir cj-dir];; formula 2 in paper
```

Comment:
calculate-base ist die funktion, die wir für alle brauchen, die basis der funktionen, die in allen drei gleichungen im paper benutzt wird